### ISSUE TITLE
Cannot add property 0, object is not extensible

### ISSUE THREAD

**Issue by natew** [rollup/rollup#5770]
### Rollup Version

4.29.0

### Operating System (or Browser)

Mac

### Link To Reproduction

https://github.com/natew/rollup-bug

### Expected Behaviour

To repro:

```
git clone git@github.com:natew/rollup-bug.git
cd rollup-bug
yarn
yarn dev
curl http://127.0.0.1:8081/index.bundle\?platform\=ios > ~/tmp/out.js
```

You'll see an error:

![CleanShot 2024-12-20 at 13 24 46@2x](https://github.com/user-attachments/assets/6ea3de94-e70d-454c-bc0d-00ddd861a0dc)

### Actual Behaviour

Error:

```
    at Array.push (<anonymous>)
    at LogicalExpression.getLiteralValueAtPath (file:///private/tmp/s2/node_modules/vite/node_modules/rollup/dist/es/shared/node-entry.js:13090:53)
    at IfStatement.getTestValue (file:///private/tmp/s2/node_modules/vite/node_modules/rollup/dist/es/shared/node-entry.js:12054:77)
    at IfStatement.include (file:///private/tmp/s2/node_modules/vite/node_modules/rollup/dist/es/shared/node-entry.js:11991:36)
```

This is due to:

```
this.expressionsToBeDeoptimized.push(origin);
```

note that `this.expressionsToBeDeoptimized.push` is set like:

```
this.expressionsToBeDeoptimized = EMPTY_ARRAY;
```

and `EMPTY_ARRAY` has `Object.freeze()`, so it's not mutable.

This started happening only in `4.29`

**Comment by TrickyPi** [comment]
Great catch! We'll fix it as soon as possible.

**Comment by lukastaegert** [comment]
Thanks for the find, here is a simple reproduction: http://localhost:5173/repl/?local=true&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmxldCUyMGNvdW50JTIwJTNEJTIwMCUzQiU1Q25sZXQlMjBmb28lMjAlM0QlMjBmYWxzZSUzQiU1Q25sZXQlMjBjb25kaXRpb24lMjAlM0QlMjBmb28lMjAlMjYlMjYlMjBmYWxzZSUzQiU1Q25pZiUyMCghY29uZGl0aW9uKSUyMGNvdW50JTJCJTJCJTNCJTVDbmZvbyUyMCUzRCUyMHRydWUlM0IlNUNuaWYlMjAoIWNvbmRpdGlvbiklMjBjb3VudCUyQiUyQiUzQiU1Q25hc3NlcnQuc3RyaWN0RXF1YWwoY291bnQlMkMlMjAyKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTdEJTdE

@TrickyPi My feeling is that in this particular case, in `deoptimizeCache` we should not assign `EMPTY_ARRAY` but rather set `this.expressionsToBeDeoptimized.length = 0`. Due to the fact that we do not always fully deoptimize logical expressions, there is a chance we might deoptimize more than once, and we still want to be able to handle that second deoptimization. With that in mind I would also suggest to change `LogicalExpression.getLiteralValueAtPath` to also run `this.expressionsToBeDeoptimized.push(origin)` when we return `UnknownTruthyValue`

**Comment by TrickyPi** [comment]
I gusse the issue is that we optimize the literval value if the usedBranch is null, even though the cache of  `logicalExpression` is deoptimzied. So I introduce the `isCacheDeoptimized` flag to fixed that in branch `fix/5770`.

**Comment by lukastaegert** [comment]
The longer I think about this, maybe a better and cleaner approach would be to only use the new logic in case the expression has not yet been deoptimized. To do so, we should no longer rely on "this.usedBranch === null` to detect if the cache was deoptimized, but maybe introduce a new, separate flag, e.g.

```js
export default class LogicalExpression extends NodeBase implements DeoptimizableEntity {
// ...
	//private hasDeoptimizedCache = false;
	private get hasDeoptimizedCache(): boolean {
		return isFlagSet(this.flags, Flag.hasDeoptimizedCache);
	}
	private set hasDeoptimizedCache(value: boolean) {
		this.flags = setFlag(this.flags, Flag.hasDeoptimizedCache, value);
	}

// ...

deoptimizeCache(): void {
		if (this.usedBranch) {
			const unusedBranch = this.usedBranch === this.left ? this.right : this.left;
			this.usedBranch = null;
			this.hasDeoptimizedCache = true; // add this
			unusedBranch.deoptimizePath(UNKNOWN_PATH);
			const {
				scope: { context },
				expressionsToBeDeoptimized
			} = this;
			this.expressionsToBeDeoptimized = EMPTY_ARRAY as unknown as DeoptimizableEntity[];
			for (const expression of expressionsToBeDeoptimized) {
				expression.deoptimizeCache();
			}
			// Request another pass because we need to ensure "include" runs again if
			// it is rendered
			context.requestTreeshakingPass();
		}
	}

// ...

getLiteralValueAtPath(
		path: ObjectPath,
		recursionTracker: PathTracker,
		origin: DeoptimizableEntity
	): LiteralValueOrUnknown {
		const usedBranch = this.getUsedBranch();
		if (usedBranch) {
			this.expressionsToBeDeoptimized.push(origin);
			return usedBranch.getLiteralValueAtPath(path, recursionTracker, origin);
		} else if (!this.hasDeoptimizedCache) { // changed the condition here
			const rightValue = this.right.getLiteralValueAtPath(path, recursionTracker, origin);
			const booleanOrUnknown = tryCastLiteralValueToBoolean(rightValue);
			if (typeof booleanOrUnknown !== 'symbol') {
				if (!booleanOrUnknown && this.operator === '&&') {
					this.expressionsToBeDeoptimized.push(origin);
					return UnknownFalsyValue;
				}
				if (booleanOrUnknown && this.operator === '||') {
					this.expressionsToBeDeoptimized.push(origin); // add this
					return UnknownTruthyValue;
				}
			}
		}
		return UnknownValue;
	}
```

**Comment by lukastaegert** [comment]
Ah perfect, forget what I wrote

**Comment by lukastaegert** [comment]
Can you make it a bit flag as I suggested? That saves some memory.

**Comment by TrickyPi** [comment]
This issue has been resolved via https://github.com/rollup/rollup/pull/5771 as part of rollup@4.29.1. You can test it via `npm install rollup`.

**Comment by genomedicsmarzoli** [comment]
I have the same error with rollup 4.30.1 using it inside Angular build

**Comment by genomedicsmarzoli** [comment]
I tried to create a minimal reproduction but I was unable to reproduce it if not with the entire project which generates more than 85 chunks.

**Comment by TrickyPi** [comment]
Can you try installing version 4.29.1 to check if the error disappears?

**Comment by TrickyPi** [comment]
If this issue is still present in v4.29.1, please check this PR https://github.com/rollup/rollup/pull/5804#issuecomment-2598525673 to see if it could resolve your problem.

**Comment by genomedicsmarzoli** [comment]
Since the problem is generated during the compilation with Angular, I used the overrides attribute of the package.json to change the nested version of rollup inside Angular. With 4.29.1 the problem persists and also with 4.29.0. While from version 4.28.1 and earlier it works.
I also tried to put the version gh-5770_try_to_fix but the problem persists.

**Comment by TrickyPi** [comment]
That is really weird. I have no idea without a minimal reproduction. Could you try adding the following code to your local Rollup code? I hope this output can help you create a shareable minimal reproduction.
```js
if (booleanOrUnknown && this.operator === '||') {
	if (Object.isFrozen(this.expressionsToBeDeoptimized)) {
		console.log(
			'unexpected: expressionsToBeDeoptimized is frozen',
			this.scope.context.module.id,
			this.start,
			this.end
		);
	}
	this.expressionsToBeDeoptimized.push(origin);
	return UnknownTruthyValue;
}
```

**Comment by genomedicsmarzoli** [comment]
I put the version back to 4.29.1 and 4.30.1 and changed the code as you told me and the console.log is executed

<img width="1389" alt="Image" src="https://github.com/user-attachments/assets/f7594177-1ec1-470f-944c-2f49ca8a1417" />

**Comment by genomedicsmarzoli** [comment]
Hello I managed to reproduce a minimal example and uploaded it to this repository.

https://github.com/genomedicsmarzoli/test-rollup

to test it:

npm install --force
npm run prod

**Comment by TrickyPi** [comment]
👍 Thanks for your efforts! I will look into this.

**Comment by TrickyPi** [comment]
Actually, the problem in your reproduction is fixed by this PR https://github.com/rollup/rollup/pull/5804. Please set the `overrides` in your package.json, re-install `node_modules` and run the build command. The error should disappear.
```json
{
  "overrides": {
    "rollup": "github:rollup/rollup#gh-5770_try_to_fix"
  }
}
```
That being said, your minimal reproduction is still very helpful. It will help us to write an effective and concise test case for this PR  https://github.com/rollup/rollup/pull/5804.

**Comment by ysy8** [comment]
This error reappeared in Rollup v4.45.0. @lukastaegert

**Comment by TrickyPi** [comment]
> This error reappeared in Rollup v4.45.0.

Could you provide a minimal reproduction? I think a fix similar to #5771 might also resolve your issue.

**Comment by katywings** [comment]
Also having this error with v4.45 and #6009 does fix it for me.

**Comment by cyb3rsalih** [comment]
@katywings  Could you give me the link you've pasted to package.json. 
I have same issue  @TrickyPi  used for another pr
`
  "overrides": {
        "rollup": "github:rollup/rollup#gh-5770_try_to_fix"
    }
`

Thanks

**Comment by katywings** [comment]
@cyb3rsalih The link? I used `rollup/rollup#fix/conditional-expression-literal-value` in the override, but that was for pnpm, I am not 100% sure but you might have to slightly adapt it for npm 🤔

**Comment by cyb3rsalih** [comment]
I did, Unfortunately got same error when build
 "overrides": {
        "rollup": "github:rollup/rollup#fix/conditional-expression-literal-value"
    }

```error during build:
Cannot add property 0, object is not extensible
    at Array.push (<anonymous>)
    at ConditionalExpression.getLiteralValueAtPath (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:12270:45)
    at UnaryExpression.getLiteralValueAtPath (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:15080:45)
    at UnaryExpression.getRenderedLiteralValue (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:15118:44)
    at UnaryExpression.include (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:15126:25)
    at UnaryExpression.include (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:15129:27)
    at LogicalExpression.include (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:14059:24)
    at LogicalExpression.include (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:14059:24)
    at ReturnStatement.include (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:14515:24)
    at BlockStatement.include (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:6519:26)```

**Comment by TrickyPi** [comment]
> I did, Unfortunately got same error when build "overrides": { "rollup": "github:rollup/rollup#fix/conditional-expression-literal-value" }
> 
> `error during build: Cannot add property 0, object is not extensible at Array.push (<anonymous>) at ConditionalExpression.getLiteralValueAtPath (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:12270:45) at UnaryExpression.getLiteralValueAtPath (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:15080:45) at UnaryExpression.getRenderedLiteralValue (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:15118:44) at UnaryExpression.include (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:15126:25) at UnaryExpression.include (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:15129:27) at LogicalExpression.include (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:14059:24) at LogicalExpression.include (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:14059:24) at ReturnStatement.include (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:14515:24) at BlockStatement.include (file:///Users/msalihb/Desktop/hatim-app/node_modules/rollup/dist/es/shared/node-entry.js:6519:26)`

Could you help check if the overrides are taking effect correctly?

**Comment by TrickyPi** [comment]
> This error reappeared in Rollup v4.45.0.

This issue should be fixed in PR #6009. If you still encounter the problem, a minimal reproduction would be greatly appreciated.

In the meantime, before a new patch release is available, you can temporarily downgrade to a version prior to 4.45.0.

**Comment by suzumejakku** [comment]
Hi,
I'm getting the same problem through vite 7.0.4 :

```
Build failed in 6.42s
error during build:
Cannot add property 0, object is not extensible
    at Array.push (<anonymous>)
    at ConditionalExpression.getLiteralValueAtPath (file:///rails/node_modules/rollup/dist/es/shared/node-entry.js:12270:45)
    at file:///rails/node_modules/rollup/dist/es/shared/node-entry.js:5064:30
    at EntityPathTracker.withTrackedEntityAtPath (file:///rails/node_modules/rollup/dist/es/shared/node-entry.js:2014:24)
    at LocalVariable.getLiteralValueAtPath (file:///rails/node_modules/rollup/dist/es/shared/node-entry.js:5062:33)
    at Identifier.getLiteralValueAtPath (file:///rails/node_modules/rollup/dist/es/shared/node-entry.js:5207:48)
    at ObjectExpression.getObjectEntity (file:///rails/node_modules/rollup/dist/es/shared/node-entry.js:14262:47)
    at ObjectExpression.deoptimizePath (file:///rails/node_modules/rollup/dist/es/shared/node-entry.js:14201:14)
    at CallExpression.deoptimizePath (file:///rails/node_modules/rollup/dist/es/shared/node-entry.js:11929:30)
    at Property.deoptimizePath (file:///rails/node_modules/rollup/dist/es/shared/node-entry.js:5667:36)

pid 52 exit 1
Build with Vite failed!
```

**Comment by ildevblu** [comment]
Same exact error here with vite 7.0.4. Problem temporary bypassed by setting `build.rollupOptions.treeshake: false` on vite.config.ts (with obvious package size downsides)

Edit: Also solved by overriding rollup version to 4.44.2 on package.json

**Comment by suzumejakku** [comment]
> Edit: Also solved by overriding rollup version to 4.44.2 on package.json

Thank you !

**Comment by alextrazah** [comment]
> Same exact error here with vite 7.0.4. Problem temporary bypassed by setting `build.rollupOptions.treeshake: false` on vite.config.ts (with obvious package size downsides)
> 
> Edit: Also solved by overriding rollup version to 4.44.2 on package.json

Thanks !!!

**Comment by hassamulhaq** [comment]
> Same exact error here with vite 7.0.4. Problem temporary bypassed by setting `build.rollupOptions.treeshake: false` on vite.config.ts (with obvious package size downsides)
> 
> Edit: Also solved by overriding rollup version to 4.44.2 on package.json


I got below error on `pnpm run build`
```
✓ 3211 modules transformed.
✗ Build failed in 4.94s
error during build:
Cannot add property 0, object is not extensible
    at Array.push (<anonymous>)
    at ConditionalExpression.getLiteralValueAtPath
```

**Thanks _ildevblu_ it fixed** by adding below code in `vite.config.ts`
```
build: {
    rollupOptions: {
        treeshake: false
    }
}
```

in package.json i have. I'm using **Laravel**.
```
"engines": {
    "pnpm": ">=9.15.0",
    "npm": "Please use pnpm instead.",
    "node": ">=22.13.1",
    "yarn": "Please use pnpm instead."
},
"dependencies": {
...
...
"@vitejs/plugin-react": "^4.3.4",
"vite": "^6.0"
},
"optionalDependencies": {
    "@rollup/rollup-linux-x64-gnu": "4.9.5",
    "@tailwindcss/oxide-linux-x64-gnu": "^4.0.1",
    "lightningcss-linux-x64-gnu": "^1.29.1"
},
"packageManager": "pnpm@10.13.1",
"pnpm": {
    "onlyBuiltDependencies": [
        "esbuild"
    ]
}
```

**Comment by Anas-Abuhatab** [comment]
> Same exact error here with vite 7.0.4. Problem temporary bypassed by setting `build.rollupOptions.treeshake: false` on vite.config.ts (with obvious package size downsides)
> 
> Edit: Also solved by overriding rollup version to 4.44.2 on package.json


Thanks its solved the issue but

build.rollupOptions.treeshake: false

Warning: Only use this in development or debugging
Disabling tree-shaking can:
Increase bundle size significantly
Include unused code that slows down the app

we need to solve the issue instead set treeshake to false 

@TrickyPi

**Comment by ildevblu** [comment]
> > Same exact error here with vite 7.0.4. Problem temporary bypassed by setting `build.rollupOptions.treeshake: false` on vite.config.ts (with obvious package size downsides)
> > Edit: Also solved by overriding rollup version to 4.44.2 on package.json
> 
> Thanks its solved the issue but
> 
> build.rollupOptions.treeshake: false
> 
> Warning: Only use this in development or debugging Disabling tree-shaking can: Increase bundle size significantly Include unused code that slows down the app
> 
> we need to solve the issue instead set treeshake to false
> 
> [@TrickyPi](https://github.com/TrickyPi)

I agree.  It's better to override the rollup dependency to the last working version. Just add this in package.json until the fix is out:


```
{
  "overrides": {
    "rollup": "4.44.2"
  }
}
```

and run npm install

**Comment by burakdzgr** [comment]
How can I pinpoint which page and which block the error is coming from? I'm working on a large-scale project, and I've been struggling for hours today because of this issue. I can't figure out exactly where it's originating. How can I pinpoint which page and which block the error is coming from? I'm working on a large-scale project, and I've been struggling for hours today because of this issue. I can't figure out exactly where it's originating. I also added the block you suggested 
`"overrides": {
  "rollup": "4.44.2"
}`
but the problem still persists.


<img width="1099" height="684" alt="Image" src="https://github.com/user-attachments/assets/03088dd5-6917-4a85-941b-be3a0d6a68fd" />

**Comment by TrickyPi** [comment]
> How can I pinpoint which page and which block the error is coming from? I'm working on a large-scale project, and I've been struggling for hours today because of this issue. I can't figure out exactly where it's originating. How can I pinpoint which page and which block the error is coming from? I'm working on a large-scale project, and I've been struggling for hours today because of this issue. I can't figure out exactly where it's originating. I also added the block you suggested 
> `"overrides": {
>   "rollup": "4.44.2"
> }`
> but the problem still persists.
> 
> 
> <img width="1099" height="684" alt="Image" src="https://github.com/user-attachments/assets/03088dd5-6917-4a85-941b-be3a0d6a68fd" />

using "resolutions" instead of "overrides" if your package manager is Yarn

**Comment by burakdzgr** [comment]
> > How can I pinpoint which page and which block the error is coming from? I'm working on a large-scale project, and I've been struggling for hours today because of this issue. I can't figure out exactly where it's originating. How can I pinpoint which page and which block the error is coming from? I'm working on a large-scale project, and I've been struggling for hours today because of this issue. I can't figure out exactly where it's originating. I also added the block you suggested
> > `"overrides": { "rollup": "4.44.2" }`
> > but the problem still persists.
> > <img alt="Image" width="1099" height="684" src="https://private-user-images.githubusercontent.com/61769761/465961459-03088dd5-6917-4a85-941b-be3a0d6a68fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI0OTIwMjQsIm5iZiI6MTc1MjQ5MTcyNCwicGF0aCI6Ii82MTc2OTc2MS80NjU5NjE0NTktMDMwODhkZDUtNjkxNy00YTg1LTk0MWItYmUzYTBkNmE2OGZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE0VDExMTUyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyYzc2OTIwM2E0YmUwNmM5ZjJkNDhmNTNlZDA2MDg2MzQ5MzI0ZGQyYmNhMDVkMGVkNzljZWQ2NDUyMTgyNjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vJgfMjuEE1m74SxdSnINEjrgy4wHcSQXFdGZprJH2fY">
> 
> using "resolutions" instead of "overrides" if your package manager is Yarn


<img width="1263" height="820" alt="Image" src="https://github.com/user-attachments/assets/d65b9b2c-c077-43c1-a566-52f49d2538e9" />

The problem is still there. I’ve tried everything, but it still doesn’t seem to be fixed.

**Comment by just-be-weird** [comment]
Yup override method worked. I had to pin it to `v4.40.0`. Thanks!

```json
  "overrides": {
    "rollup": "4.40.0"
  }
```

**Comment by TrickyPi** [comment]
> > > How can I pinpoint which page and which block the error is coming from? I'm working on a large-scale project, and I've been struggling for hours today because of this issue. I can't figure out exactly where it's originating. How can I pinpoint which page and which block the error is coming from? I'm working on a large-scale project, and I've been struggling for hours today because of this issue. I can't figure out exactly where it's originating. I also added the block you suggested
> > > `"overrides": { "rollup": "4.44.2" }`
> > > but the problem still persists.
> > > <img alt="Image" width="1099" height="684" src="https://private-user-images.githubusercontent.com/61769761/465961459-03088dd5-6917-4a85-941b-be3a0d6a68fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI0OTIwMjQsIm5iZiI6MTc1MjQ5MTcyNCwicGF0aCI6Ii82MTc2OTc2MS80NjU5NjE0NTktMDMwODhkZDUtNjkxNy00YTg1LTk0MWItYmUzYTBkNmE2OGZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE0VDExMTUyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyYzc2OTIwM2E0YmUwNmM5ZjJkNDhmNTNlZDA2MDg2MzQ5MzI0ZGQyYmNhMDVkMGVkNzljZWQ2NDUyMTgyNjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vJgfMjuEE1m74SxdSnINEjrgy4wHcSQXFdGZprJH2fY">
> > 
> > 
> > using "resolutions" instead of "overrides" if your package manager is Yarn
> 
> <img alt="Image" width="1263" height="820" src="https://private-user-images.githubusercontent.com/61769761/465986104-d65b9b2c-c077-43c1-a566-52f49d2538e9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI1NDM1NDUsIm5iZiI6MTc1MjU0MzI0NSwicGF0aCI6Ii82MTc2OTc2MS80NjU5ODYxMDQtZDY1YjliMmMtYzA3Ny00M2MxLWE1NjYtNTJmNDlkMjUzOGU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE1VDAxMzQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYyZDkwN2E4NzI2MTFmYzZkYmU5NjYzNDBjZTRmMTAyNDQ2ZTM2MTcwZmVkZTM4MzU4MDcxNWEwMmMzZTVmZjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.YJCfGzNPbcl2n47grkgcfex1CVAEJSiAwZh_C19kS0M">
> The problem is still there. I’ve tried everything, but it still doesn’t seem to be fixed.

Before building, you can run [yarn why](https://yarnpkg.com/cli/why) to check if your "resolutions" setting is actually working. If Rollup is pinned to 4.44.2 and the problem still exists, your issue might be a different one.

**Comment by Kynetic25** [comment]
> Yup override method worked. I had to pin it to `v4.40.0`. Thanks!
> 
>   "overrides": {
>     "rollup": "4.40.0"
>   }

Yes... worked for me. Added to package.json, deleted node_modules and ran npm install again. Worked really well.

**Comment by suzumejakku** [comment]
For those following this closed thread, the latest version of rollup 4.45.1 fixed it for me.
